src/main/java/net/consensys/orion/http/handler/receive/ReceiveHandler.java

```diff
@@ -104,4 +101,18 @@ public void handle(final RoutingContext routingContext) {
       routingContext.response().end(toReturn);
     });
   }
+
+  private byte[] decryptPayload(final List<Box.PublicKey> recipients, final EncryptedPayload encryptedPayload) {
+    byte[] decryptedPayload = null;
+    for (final Box.PublicKey recipient : recipients) {
+      try {
+        decryptedPayload = enclave.decrypt(encryptedPayload, recipient);
```

Comment:
Could just return the decrypted value instead of assigning and breaking out of the loop. And then returning on line 116 if nothing found.